### PR TITLE
linear iteration growth for celer and blitz

### PR DESCRIPTION
**Pull request by mathurinm**

### CODE REVIEW

**Comment by agramfort** [comment]
CI complains

**Comment by mathurinm** [comment]
It seems `get_next` must be a staticmethod
https://github.com/benchopt/benchmark_lasso/runs/4569231523?check_suite_focus=true#step:6:404

The code works so I'm not sure this check is necessary, and I'm missing why we want to impose that on `get_next` 
@tomMoral can you explain it to me?

**Comment by tomMoral** [comment]
When I implemented `get_next` in the solver, `StoppingCriterion.get_runner_instance` was called before the solver was instanciated, and thus `get_next` needed to be a static method in order to work. Since then, the runner has been re-ordered and I think this is not necessary anymore.

I guess we could drop this requirement but not sure if we should. The benefit I see in having a static method is that it enforces the fact that `get_next` should not rely in internal state of the solver. In the future, if we decide to better parallelize for solvers that do not have `callback`, this would be useful and we won't have to deprecate the use of non-static `get_next`.
WDYT @mathurinm

**Comment by mathurinm** [comment]
Since you see a potential usecase in the future, I agree to keep it static!

File: solvers/blitz.py [hunk 1]
```diff
@@ -33,6 +33,10 @@ def set_objective(self, X, y, lmbd, fit_intercept):
         blitzl1.set_tolerance(0)
         self.problem = blitzl1.LassoProblem(self.X, self.y)
 
+    @staticmethod
+    def get_next(self, previous):
```
Comment (tomMoral) [inline solvers/blitz.py:37]:
```suggestion
    def get_next(previous):
        "Linear growth for n_iter"
```

File: solvers/celer.py [hunk 2]
```diff
@@ -38,6 +38,10 @@ def run(self, n_iter):
         self.lasso.max_iter = n_iter
         self.lasso.fit(self.X, self.y)
 
+    @staticmethod
+    def get_next(self, previous):
```
Comment (tomMoral) [inline solvers/celer.py:42]:
```suggestion
    def get_next(previous):
            "Linear growth for n_iter."
```

**Comment by tomMoral** [comment]
@mathurinm can you show the result of the benchmark here?

**Comment by mathurinm** [comment]
pinging my joker @qb3

**Pull request merged**